hack/tests/scaffolding/scaffold-memcached.go

```diff
@@ -0,0 +1,285 @@
+			// stub go.mod into the local SDK repo referred to in
+			// memcached-operator's go.mod, which allows go to recognize
+			// the local SDK repo as a module.
+			sdkModPath := filepath.Join(filepath.FromSlash(replace.repo), "go.mod")
+			if _, err = os.Stat(sdkModPath); err != nil && os.IsNotExist(err) {
+				err = ioutil.WriteFile(sdkModPath, []byte("module "+sdkRepo), fileutil.DefaultFileMode)
+				if err != nil {
+					log.Fatalf("Failed to write main repo go.mod file: %v", err)
+				}
+			}
+		}
+		modBytes, err := insertGoModReplace(sdkRepo, replace.repo, replace.ref)
+		if err != nil {
+			log.Fatalf("Failed to insert go.mod replace: %v", err)
+		}
+		log.Printf("go.mod: %v", string(modBytes))
+	}
+
+	cmdOut, err = exec.Command("go", "build", "./...").CombinedOutput()
+	if err != nil {
+		log.Fatalf("Command \"go build ./...\" failed after modifying go.mod: %v\nCommand Output:\n%v", err, string(cmdOut))
+	}
+
+	// Set replicas to 2 to test leader election. In production, this should
+	// almost always be set to 1, because there isn't generally value in having
+	// a hot spare operator process.
+	opYaml, err := ioutil.ReadFile("deploy/operator.yaml")
+	if err != nil {
+		log.Fatalf("Could not read deploy/operator.yaml: %v", err)
+	}
+	newOpYaml := bytes.Replace(opYaml, []byte("replicas: 1"), []byte("replicas: 2"), 1)
+	err = ioutil.WriteFile("deploy/operator.yaml", newOpYaml, 0644)
+	if err != nil {
+		log.Fatalf("Could not write deploy/operator.yaml: %v", err)
+	}
+
+	cmd := exec.Command("operator-sdk",
+		"add",
+		"api",
+		"--api-version=cache.example.com/v1alpha1",
+		"--kind=Memcached")
+	cmd.Env = os.Environ()
+	cmdOut, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
+	}
+	cmdOut, err = exec.Command("operator-sdk",
+		"add",
+		"controller",
+		"--api-version=cache.example.com/v1alpha1",
+		"--kind=Memcached").CombinedOutput()
+	if err != nil {
+		log.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
+	}
+
+	tmplFiles := map[string]string{
+		filepath.Join(localSDKPath, "example/memcached-operator/memcached_controller.go.tmpl"): "pkg/controller/memcached/memcached_controller.go",
+		filepath.Join(localSDKPath, "test/e2e/_incluster-test-code/main_test.go"):              "test/e2e/main_test.go",
+		filepath.Join(localSDKPath, "test/e2e/_incluster-test-code/memcached_test.go"):         "test/e2e/memcached_test.go",
+	}
+	for src, dst := range tmplFiles {
+		if err := os.MkdirAll(filepath.Dir(dst), fileutil.DefaultDirFileMode); err != nil {
+			log.Fatalf("Could not create template destination directory: %s", err)
+		}
+		cmdOut, err = exec.Command("cp", src, dst).CombinedOutput()
+		if err != nil {
+			log.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
+		}
+	}
+
+	memcachedTypesFile, err := ioutil.ReadFile("pkg/apis/cache/v1alpha1/memcached_types.go")
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Nit:
```suggestion
		log.Fatalf("Could not read pkg/apis/cache/v1alpha1/memcached_types.go: %v", err)
```
Or make that path string a variable.